design/button.behavior.md

```diff
@@ -0,0 +1,55 @@
+# Button Behavior
+
+Button behavior is the high level representation of how an MRE actor should behave as a button.  This behavior consists of two actions
+that occur on the object when it is targeted and when it is activated or clicked.  As part of these interactions, the point of targeting
+of the button will be passed along as additional data to the app for consumption.
+
+Button Behavior Actions:
+- `hover` - The action that occurs when the button is targeted by the host app input system. 
+    - Hover states: `enter`, `hovering`, `exit`
+- `click` - The action that occurs when a complete click has happened on the button consisting of both a button pressed and button released event. 
+    - Click states: None.  This is a single atomic event.
+- `button` - The action that occurs when the button interaction state has changed from being pressed and then released.
+    - Button states: `pressed`, `holding`, `released`
+
+## Architecture
+
+### ButtonEventData
+``` ts
+interface ButtonEventData {
+	points: TransformLike[];
```

Comment:
I had imagined that this would be in app coordinates, but we could supply both if you think it is useful, rather than the developer need to convert them.  What are your thoughts?

Reply:
The whiteboard app will want local coordinates. It would be good to send both, or send local and make it easy to convert to app.

---
In reply to: [420549433](https://github.com/microsoft/mixed-reality-extension-sdk/pull/573#discussion_r420549433) [](ancestors = 420549433)